frontend/src/components/Widgets/CustomEntitiesDropdown.tsx

```diff
@@ -402,468 +466,443 @@ export default function CustomEntitiesDropdown({
-            display: "flex",
-            flexDirection: "column",
-            zIndex: 9999,
-          }}
-        >
-          {header?.values?.length > 0 && (
-            <Box
-              display="flex"
-              alignItems="center"
-              flexWrap="wrap"
-              gap={1}
-              sx={{
-                borderBottom: `0.0625rem solid ${popperBorderColor}`,
-                height: autocompleteOptions.length > 0 ? "2.75rem" : "auto",
-                padding:
-                  autocompleteOptions.length > 0 ? "0 0.875rem" : "0.875rem",
-              }}
-            >
-              <Typography variant="body2">{header?.label}</Typography>
-              {header?.values?.map((item: any, index: number) => (
-                <Tooltip title={item} placement="top" arrow>
-                  <Chip
-                    key={item?.id}
-                    sx={{
-                      ...styles.chip,
-                      display: "flex",
-                      textAlign: "left",
-                    }}
-                    variant="outlined"
-                    label={
-                      <>
-                        <Typography
-                          sx={{
-                            verticalAlign: "text-bottom",
-                            display: "inline-block",
-                            mr: "0.25rem",
-                            borderRadius: "0.1875rem",
-                            background: dropdownChipColor,
-                            px: "0.25rem",
-                            fontSize: "0.75rem",
-                            color: buttonOutlinedColor,
-                            fontWeight: 600,
-                            height: "1.125rem",
-                          }}
-                          component="span"
-                        >
-                          {index + 1}
-                        </Typography>
-                        {item}
-                      </>
-                    }
-                  />
-                </Tooltip>
-              ))}
-            </Box>
-          )}
-          <Box
-            display="flex"
-            flex={1}
-            height={
-              autocompleteOptions.length > 0 ? "calc(100% - 2.75rem)" : "auto"
-            }
-          >
-            <Box
-              sx={{
-                ...styles.list,
-                width: autocompleteOptions.length > 0 ? "50%" : "100%",
-              }}
+        {
+          disabled ? <CommonChipBox
+              selectedOptions={selectedOptions}
+              CustomInputChip={CustomInputChip}
+              styles={styles}
+              disabled={disabled}
+              handleChipRemove={handleChipRemove}
+              chipsNumber={chipsNumber}
+            />: <>
+            <Badge
+              sx={{...styles.badge, flex: 1}}
+              badgeContent={
+                !isAllSelectedValuesFromTheAboveLayer ? 0 : selectedOptions?.length
```

Comment:
nitpick (non-blocking): isAllSelectedValuesFromTheAboveLayer should be named areAllSelectedValuesFromTheAboveLayer